tests/pytests/unit/utils/test_network.py

```diff
@@ -1,8 +1,1547 @@
+    ), patch("socket.getfqdn", MagicMock(return_value="ip6-localhost")), patch(
+        "socket.getaddrinfo",
+        MagicMock(return_value=[(2, 3, 0, "localhost", ("127.0.1.1", 0))]),
+    ), patch(
+        "salt.utils.files.fopen", mock_open()
+    ), patch(
+        "salt.utils.network.ip_addrs",
+        MagicMock(return_value=["127.0.0.1", "::1", "fe00::0", "fe02::1", "1.2.3.4"]),
+    ):
+        assert network.generate_minion_id() == "1.2.3.4"
+
+
+def test_gen_mac():
+    with patch("random.randint", return_value=1) as random_mock:
+        assert random_mock.return_value == 1
+        ret = network.gen_mac("00:16:3E")
+        expected_mac = "00:16:3E:01:01:01"
+        assert ret == expected_mac
+
+
+def test_mac_str_to_bytes():
+    pytest.raises(ValueError, network.mac_str_to_bytes, "31337")
+    pytest.raises(ValueError, network.mac_str_to_bytes, "0001020304056")
+    pytest.raises(ValueError, network.mac_str_to_bytes, "00:01:02:03:04:056")
+    pytest.raises(ValueError, network.mac_str_to_bytes, "a0:b0:c0:d0:e0:fg")
+    assert b"\x10\x08\x06\x04\x02\x00" == network.mac_str_to_bytes("100806040200")
+    assert b"\xf8\xe7\xd6\xc5\xb4\xa3" == network.mac_str_to_bytes("f8e7d6c5b4a3")
+
+
+@pytest.mark.slow_test
+def test_generate_minion_id_with_long_hostname():
+    """
+    Validate the fix for:
+
+    https://github.com/saltstack/salt/issues/51160
+    """
+    long_name = "localhost-abcdefghijklmnopqrstuvwxyz-abcdefghijklmnopqrstuvwxyz"
+    with patch("socket.gethostname", MagicMock(return_value=long_name)):
+        # An exception is raised if unicode is passed to socket.getfqdn
+        minion_id = network.generate_minion_id()
+    assert minion_id != ""
+
+
+def test_filter_by_networks_with_no_filter():
+    ips = ["10.0.123.200", "10.10.10.10"]
+    with pytest.raises(TypeError):
+        network.filter_by_networks(ips)  # pylint: disable=no-value-for-parameter
+
+
+def test_filter_by_networks_empty_filter():
+    ips = ["10.0.123.200", "10.10.10.10"]
+    assert network.filter_by_networks(ips, []) == []
+
+
+def test_filter_by_networks_ips_list():
+    ips = [
+        "10.0.123.200",
+        "10.10.10.10",
+        "193.124.233.5",
+        "fe80::d210:cf3f:64e7:5423",
+    ]
+    networks = ["10.0.0.0/8", "fe80::/64"]
+    assert network.filter_by_networks(ips, networks) == [
+        "10.0.123.200",
+        "10.10.10.10",
+        "fe80::d210:cf3f:64e7:5423",
+    ]
+
+
+def test_filter_by_networks_interfaces_dict():
+    interfaces = {
+        "wlan0": ["192.168.1.100", "217.5.140.67", "2001:db8::ff00:42:8329"],
+        "eth0": [
+            "2001:0DB8:0:CD30:123:4567:89AB:CDEF",
+            "192.168.1.101",
+            "10.0.123.201",
+        ],
+    }
+    assert network.filter_by_networks(
+        interfaces, ["192.168.1.0/24", "2001:db8::/48"]
+    ) == {
+        "wlan0": ["192.168.1.100", "2001:db8::ff00:42:8329"],
+        "eth0": ["2001:0DB8:0:CD30:123:4567:89AB:CDEF", "192.168.1.101"],
+    }
```

Comment:
```suggestion
    ret = network.filter_by_networks(
        interfaces, ["192.168.1.0/24", "2001:db8::/48"]
    )
    expected = {
        "wlan0": ["192.168.1.100", "2001:db8::ff00:42:8329"],
        "eth0": ["2001:0DB8:0:CD30:123:4567:89AB:CDEF", "192.168.1.101"],
    }
    assert ret == expected
```
Doesn't this read better?

Reply:
yes, but I usually put expected before the testing call
Done

Reply:
> yes, but I usually put expected before the testing call
That's the wrong way.

Which one shows you that you got one more key that you were expecting? https://gist.github.com/s0undt3ch/92d065815f4ce526345f72d40a40074d#file-pytest-output-txt